Application/Published/Upgrade-AdminApp/Deploy-EdFiOdsAdminApp.ps1

```diff
@@ -0,0 +1,220 @@
+#>
+Param(
+
+	[string] 
+	[Parameter(Mandatory=$true)] 
+	$ResourceGroupName,
+
+	[string] 
+	[Parameter(Mandatory=$true)] 
+	$AdminAppName,
+
+	[ValidateSet("East US", "South Central US")]
+	[string] 
+	[Parameter(Mandatory=$true)]	
+	$AppInsightLocation,
+	
+	[string]
+	[Parameter(Mandatory=$true)]
+	$ProductionApiUrl,
+
+	[string]
+	[Parameter(Mandatory=$true)]
+	$SQLServerHostname,
+
+	[string]
+	[Parameter(Mandatory=$true)]
+	$SQLServerUserName,
+
+	[SecureString]
+	[Parameter(Mandatory=$true)]
+	$SQLServerPassword,
+
+	[string]
+	[Parameter(Mandatory=$false)] 
+	$EncryptionKey,
+
+	[string] 
+	[Parameter(Mandatory=$false)] 
+	$TemplateFileDirectory = '$PSScriptRoot',
+
+	[string]
+	[Parameter(Mandatory=$false)]
+	$AdminAppVersion = '2.2.0',
+	
+	[string]
+	[ValidateSet('Release','Test')]
+	$Edition = 'Release'	
+)
+
+Import-Module $PSScriptRoot\..\Dependencies.psm1 -Force -DisableNameChecking
+Import-Module $PSScriptRoot\..\EdFiOdsDeploy.psm1 -Force -DisableNameChecking
+
+Use-Module "AzureRM" "4.3.1"
+Use-Module "AzureRM.profile" "3.3.1"
+
+# Powershell doesn't support using $PSScriptRoot in parameter defaults,
+# so using that value as a marker to update with the real PSScriptRoot value
+if ($TemplateFileDirectory -eq '$PSScriptRoot')
+{
+	$TemplateFileDirectory = $PSScriptRoot;
+}
+
+$CacheTimeOut = "10"
+
+$AdminAppTemplateFile = "$TemplateFileDirectory\OdsAdminApp.Upgrade.json"
+$AdminAppTemplateParametersFile = "$TemplateFileDirectory\OdsAdminApp.parameters.Upgrade.json"
+
+function Deploy-AdminApp()
+{
+	if(-not $EncryptionKey)
+	{
+		$aes = [System.Security.Cryptography.Aes]::Create()
+		$aes.KeySize = 256
+		$aes.GenerateKey()
+		$EncryptionKey = [System.Convert]::ToBase64String($aes.Key)
+	}
+	
+	$deployParameters = New-Object -TypeName Hashtable
+	$deployParameters.Add("version", $AdminAppVersion)
+	$deployParameters.Add("edition", $Edition)
+	$deployParameters.Add("appInsightsLocation", $AppInsightLocation)
+	$deployParameters.Add("odsInstanceName", $ResourceGroupName)
+	$deployParameters.Add("sqlServerAdminLogin", $SQLServerUserName)
+	$deployParameters.Add("sqlServerAdminPassword", $SQLServerPassword)
+	$deployParameters.Add("productionApiUrl", $ProductionApiUrl)	
+	$deployParameters.Add("metadataCacheTimeOut", $CacheTimeOut)
+	$deployParameters.Add("adminAppNameToDeploy", $AdminAppName)
+	$deployParameters.Add("encryptionKey", $EncryptionKey)
+	
+	$templateFile = $AdminAppTemplateFile
+	$templateParametersFile = $AdminAppTemplateParametersFile
+
+	Try
+	{
+		Write-Host "Deploying ODS Admin App"		
+		$deploymentResult = New-AzureRmResourceGroupDeployment -DeploymentDebugLogLevel All -Name ((Get-ChildItem $templateFile).BaseName + '-' + ((Get-Date).ToUniversalTime()).ToString('MMdd-HHmm')) `
+									   -ResourceGroupName $ResourceGroupName `
+									   -TemplateFile $templateFile `
+									   -TemplateParameterFile $templateParametersFile `
+									   @deployParameters `
+									   -Force -Verbose -ErrorAction Stop
```

Comment:
Here's a possible formatting improvement. However, there's actually a better approach: why not move these other parameters into $deployParameters so that they're all defined in one place? For `-Force` and `-Verbose`, use `Force = $True` and `Verbose = $True` when defining the [splat](https://devblogs.microsoft.com/scripting/use-splatting-to-simplify-your-powershell-scripts/).

```suggestion
                $name = ((Get-ChildItem $templateFile).BaseName + '-' + ((Get-Date).ToUniversalTime()).ToString('MMdd-HHmm'))
		$deploymentResult = New-AzureRmResourceGroupDeployment `
                                  -DeploymentDebugLogLevel All `
                                  -Name $name `
                                  -ResourceGroupName $ResourceGroupName `
                                  -TemplateFile $templateFile `
                                  -TemplateParameterFile $templateParametersFile `
                                  @deployParameters `
                                  -Force `
                                  -Verbose `
                                  -ErrorAction Stop
```